src/Shapes.hs

```diff
@@ -780,16 +673,25 @@ data TypeIn a = TNum a
               | TBool Bool
               | TStr String
               | TInt Integer
+              | TFloat Float
               | TPt (Pt2 a)
               | TPath [Pt2 a]
-              | TArrow SolidArrow
               | TColor Color
-              | TStyle String -- dotted, etc.
+              | TStyle String -- | dotted, etc.
+              | TAllShapes String -- | Substance ID
```

Comment:
kind of a confusing name, can we call this something like `TSubObjShapes`?

Reply:
Now that I think about it, I don't see any motivation for having this shortcut to access all objects. All other parts of this feature guarantee correct ordering, but if you have something like `func(A, B.circ)`, there's no way to tell how many arguments this function is taking without looking at the Style block(s) for `A`. Do you think we should still keep this feature?

Reply:
I agree with the ordering/argument problems. Let's remove the "GPI synonym feature" for now.